### PR TITLE
[PWN-7078] Refactor AnalyticsManager

### PR DESCRIPTION
**Pull request by bigearsenal**
## Link jira to issue
https://p2pvalidator.atlassian.net/browse/PWN-7078

### `Uppercased_Snake_Case` problem
For iOS development, it is standard and easier to use `camelCase` than `Uppercased_Snake_Case`, but the implementation of our Data analysts requires that event names and params has to be written in `Uppercased_Snake_Case`, for example:  eventName = "User_Has_Positive_Balance", params: ["Is_Positive": true].
So in p2p-wallet-ios we use auto conversion from `camelCase` to `Uppercased_Snake_Case` by [mirroring](https://www.avanderlee.com/swift/reflection-how-mirror-works/) `AnalyticsEvent` enum and get the `Uppercased_Snake_Case` automatically.
It works well in almost all cases. But sometimes we needs to custom the name and params of the event base on the requirement of Analyst team. For example `isSellEnabled` has to be sent instead of standard `Is_Sell_Enabled`.
Solution: give developers ability to modify the event name and params if needed.

```swift
/// The name of the event to send
var eventName: String? {
    // By default, name of the event will be converted from `camelCase` to `Uppercased_Snake_Case` format
    // For example: `KeyAppAnalyticsEvent.mainScreenSwapOpen` will be converted to "Main_Screen_Swap_Open" automatically.
    // Example: modify the name manually and prevent default behavior
    // switch self {
    // case .login:
    //     return "UserLoggedIn"
    // default:
    //     break
    // }
    
    // Default converter from `camelCase` to `Uppercased_Snake_Case` format
    return mirror.label.snakeAndFirstUppercased
}

/// Params sent with event
var params: [String: Any]? {
    guard !mirror.params.isEmpty else { return nil }
    
    // The same for params, params key & value can be customized too, if not, it will be automatically converted to `Uppercased_Snake_Case`
    
    // Example: modify the key & value manually and prevent default behavior if needed
    // switch self {
    // case let .login(userId: String):
    //     return ["Id": userId] // by default userId -> User_Id
    // default:
    //     break
    // }
    
    // Default converter from `camelCase` to `Uppercased_Snake_Case` format
    let formatted = mirror.params.map { ($0.key.snakeAndFirstUppercased ?? "", $0.value) }
    return Dictionary(uniqueKeysWithValues: formatted)
}
```
### Multiple providers problem
Some events have to be sent to multiple providers, some events not.
Solution: add parameters `providerIds` for each event and indicate the list of providers that event can be sent to.

```swift
/// Array of sending providers, even will be sent to only these defined providers
var providerIds: [AnalyticsProviderId] {
    // By default, all events will be sent to amplitude only
    var ids: [KeyAppAnalyticsProviderId] = [
        .amplitude
    ]
    
    // for some events, we will sent to appsFlyer and firebaseAnalytics
    switch self {
    case .onboardingStartButton,
            .creationPhoneScreen,
            .createSmsValidation,
            .createConfirmPin,
            .usernameCreationScreen,
            .usernameCreationButton,
            .restoreSeed,
            .onboardingMerged,
            .login,
            .buyButtonPressed,
            .sendNewConfirmButtonClick,
            .swapClickApproveButton:
        ids.append(contentsOf: [
            .appsFlyer,
            .firebaseAnalytics
        ])
    default:
        break
    }
    return ids.map(\.rawValue)
}
```
### Diversity between AnalyticsService and AnalyticsManager
There is 2 service that works similar to each other AnalyticsService and AnalyticsManager.
Solution: Remove AnalyticsService and use AnalyticsManager with custom providers only.
## Instruction to log an event
- Step 1: Add camelCased event into `KeyAppAnalyticsEvent.swift` (with parameters if needed). Example: For event `Button_Login_Tapped` with parameters `User_Id`, `Date`, `Number_Of_Attempt`, add this code `case buttonLoginTapped(userId: String, date: Date, numberOfAttempt: Int)`.
- Step 2: Modify `eventName`, `params`, if they are not in right `Uppercased_Snake_Case` format , in `KeyAppAnalyticsEvent+Extensions.swift`.
- Step 3: Modify where to sent (providers) in `KeyAppAnalyticsEvent+Extensions.swift` if needed.
- Step 4: `@Injected private var analyticsManager: AnalyticsManager` and `analyticsManager.log(…)`

### CODE REVIEW

**Review by qwerkly — DISMISSED** [review]

File: p2p_wallet/Analytics/KeyAppAnalyticsEvent+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import Foundation
+import AnalyticsManager
+
+extension KeyAppAnalyticsEvent: MirrorableEnum {
+
+    /// The name of the event to send
+    var eventName: String? {
```
Comment (sidorov-panda) [inline p2p_wallet/Analytics/KeyAppAnalyticsEvent+Extensions.swift:7]: name instead of eventName?

Comment (bigearsenal) [reply p2p_wallet/Analytics/KeyAppAnalyticsEvent+Extensions.swift:7]: yes. Even better. Thanks.

File: p2p_wallet/Coordinators/TabBar/HomeCoordinator.swift [hunk 2]
```diff
@@ -118,8 +118,8 @@ final class HomeCoordinator: Coordinator<Void> {
             }
         case .receive(let publicKey):
             let coordinator = ReceiveCoordinator(navigationController: navigationController, pubKey: publicKey)
-            analyticsManager.log(event: AmplitudeEvent.mainScreenReceiveOpen)
-            analyticsManager.log(event: AmplitudeEvent.receiveViewed(fromPage: "main_screen"))
+            analyticsManager.log(event: .mainScreenReceiveOpen)
+            analyticsManager.log(event: .receiveViewed(fromPage: "main_screen"))
```
Comment (lisemyon) [inline p2p_wallet/Coordinators/TabBar/HomeCoordinator.swift:122]: it is interesting that somewhere we use "Main_screen" and here it is "main_screen". I wonder if we can extract this into enum or smth like that

**Review by lisemyon — DISMISSED** [review]

**Review by sidorov-panda — APPROVED** [review]

File: p2p_wallet/Coordinators/TabBar/HomeCoordinator.swift (hunk 2 above)
Comment (bigearsenal) [reply p2p_wallet/Coordinators/TabBar/HomeCoordinator.swift:122]: Yes. Let's create another task and discuss it with data analyst team later to make it more standard

**Pull request merged**